### PR TITLE
Fix bugs found when using venia-upward.yml:

### PR DESCRIPTION
**Pull request by bbatsche**
- Cleanup composer.json
- Correctly detect MIME types in Directory resolver
- Update @magento/upward-spec
- Fix definition loop detection
- Fix handling non-scalar values in Conditional resolver
- Ignore more SSL certs problems
- Add support for ignoreSSLErrors in Service resolver
- Set Content-type to application/json by default in Service resolver
- Revise handling of provided variables in Template resolver
- Make DefinitionIterator tests more inline with real world scenarios

### CODE REVIEW

File: src/Resolver/Service.php [hunk 1]
```diff
@@ -49,17 +49,30 @@ public function resolve($definition)
...
+        $headers = ['Content-type' => 'application/json'];
+
+        $url             = $this->getIterator()->get('url', $definition);
+        $query           = $this->getIterator()->get('query', $definition);
+        $method          = $definition->has('method') ? $this->getIterator()->get('method', $definition) : 'POST';
+        $variables       = $definition->has('variables') ? $this->getIterator()->get('variables', $definition) : [];
+        $ignoreSSLErrors = $definition->has('ignoreSSLErrors')
+            ? $this->getIterator()->get('ignoreSSLErrors', $definition)
+            : false;
+        $headers = $definition->has('headers')
```
Comment (tjwiebell) [inline src/Resolver/Service.php]: We should move applying this header down to the method logic gate below; only apply to POST requests.

From a spec standpoint, I think I would rather see this in config, but I suppose this reduces boilerplate, which is one of the tailing objectives at the end of the spec. Fine leaving for now.

File: composer.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
     "license": "OSL-3.0",
     "require": {
         "php": ">= 7.1",
-        "guzzlehttp/guzzle": "^6.2",
         "mustache/mustache": "^2.12",
+        "ralouphie/mimey": "^2.0",
```
Comment (tjwiebell) [inline composer.json]: Update/add `composer.lock` too

Comment (bbatsche) [reply composer.json]: `composer.log` isn't tracked with the repo

File: src/Resolver/Service.php (hunk 1 above)
Comment (bbatsche) [reply src/Resolver/Service.php]: I totally agree on preferring an application wide config over a request-by-request setup we have now. @zetlen and I discussed this briefly as well last week.

**Pull request merged**